src/managed_data/api.py

```diff
@@ -421,17 +426,36 @@ def _type_and_cleanup_dataframe(
     for column in columns:
         column_mapper[column["id"]] = column["name"]
 
-    # special treatment of Select columns
     for column in columns:
         col_id = column["id"]
 
+        print(col_id)
+        print(column["type"])
+
+        if column["type"] == "date":
+            df[col_id] = pd.to_datetime(df[col_id])
+
+        # special treatment for string columns
+        if column["type"] in ["file", "text"]:
+            df[col_id] = df[col_id].astype("string")
+
+        if column["type"] == "boolean":
+            df[col_id] = df[col_id].astype("boolean")
+
+        # special treatment of Select columns
         if column["type"] == "select" and column["cardinality"] == "one":
             categories = column["configSelect"]["options"]
             df[col_id] = pd.Categorical(df[col_id], categories=categories)
 
     # rename columns
     df = df.rename(columns=column_mapper)
 
+    # special treatment for validation status
+    categories = set(df["Validation Status"])
+    df["Validation Status"] = pd.Categorical(
+        df["Validation Status"], categories=categories
+    )
+
```

Comment:
this should work for any validation status, without needing to know what they are